src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5256,9 +5256,8 @@ public static VarHandle collectCoordinates(VarHandle target, int pos, MethodHand
         }
 
         List<Class<?>> newCoordinates = new ArrayList<>(targetCoordinates);
```

Comment:
This was a bug I discovered in the collectCoordinates implementation - the old coordinate was not removed - leading to WrongMethodTypeException.